torch/csrc/autograd/variable.h

```diff
@@ -154,7 +151,7 @@ struct VariableViewImpl : public VariableImpl {
 
   // Called after in-place modifications. Modifies the grad_fn of the base
   // Variable.
-  virtual void rebase_grad_fn(std::shared_ptr<Function> grad_fn) override;
+  void rebase_history(VarFlags flags, int output_nr, std::shared_ptr<Function> grad_fn);
```

Comment:
Maybe make this function private (and friend `Variable`) and mention in the comment that In general it should only be called via `Variable::rebase_history`?

Reply:
I don't see a reason why this should be private any more than any of the other functions and fields of the Impl classes. You should generally not access the impl classes directly, but it is inconvenient to mark them as private.

For this function, there is no difference if you call it on the impl or call the corresponding function on the Variable struct.